### ISSUE TITLE
Support time_t

### ISSUE THREAD

**Issue by nox** [rust-lang-nursery/rust-bindgen#1188]
`bindgen` currently has no knowledge of `libc::time_t`, and will thus produce this on Darwin, for example:

```rust
pub type __darwin_time_t = libc::c_long;
pub type time_t = __darwin_time_t;
```

**Comment by fitzgen** [comment]
The usual way to do this is to blacklist `time_t` and then add `pub type time_t = libc::time_t;` as a `raw_line`.

**Comment by nox** [comment]
Yeah I know that, but that requires also blacklisting `__darwin_time_t` on macOS, and it would be nice if bindgen could just do the right thing out of the box. I understand that it's not as trivial to do it, especially given `std::os::raw` doesn't even define a `time_t` type.

**Comment by kulp** [comment]
I have seen this on my Mac, too -- but it is not clear to me that the "right" thing is to add cleverness about `time_t` to `bindgen`. After all, `bindgen` is faithfully representing the behavior that the C headers are providing. Is that not the "right" thing ?

Basically, what is the value proposition here ?
